src/main/java/com/ark/inflearnback/domain/security/filter/RestLoginProcessingFilter.java

```diff
@@ -0,0 +1,44 @@
+package com.ark.inflearnback.domain.security.filter;
+
+import com.ark.inflearnback.domain.member.dto.SignRequestDto;
+import com.ark.inflearnback.domain.security.token.RestAuthenticationToken;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.util.StringUtils;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class RestLoginProcessingFilter extends AbstractAuthenticationProcessingFilter {
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    public RestLoginProcessingFilter() {
+        super(new AntPathRequestMatcher("/api/v1/login"));
+    }
+
+    @Override
+    public Authentication attemptAuthentication(final HttpServletRequest request, final HttpServletResponse response) throws AuthenticationException, IOException, ServletException {
+        if (!isRest(request)) {
+            throw new IllegalStateException("REST 요청이 아닙니다.");
+        }
+
+        final SignRequestDto memberDto = objectMapper.readValue(request.getReader(), SignRequestDto.class);
+
+        if (!StringUtils.hasText(memberDto.getEmail()) || !StringUtils.hasText(memberDto.getPassword())) {
+            throw new IllegalStateException("이메일이나 비밀번호가 입력되지 않았습니다.");
+        }
+
+        return getAuthenticationManager().authenticate(new RestAuthenticationToken(memberDto.getEmail(), memberDto.getPassword()));
+    }
+
+    private boolean isRest(final HttpServletRequest request) {
+        return "application/json".equals(request.getHeader("Content-Type"));
```

Comment:
비교하는 문자열이 `request.getHeader` 의 content-type을 json인지 확인하는거니까
```suggestion
        return request.getHeader("Content-Type").equals("application/json"));
```
이 더 나을 것 같은데 민철님의 생각이 궁금합니다.

Reply:
`request.getHeader("application/json")` 가 `null`일 경우 `NPE`가 발생합니다.
이 방식은 보통 NPE를 회피하기 위해 `equals`를 호출하는 주체가 문자열 리터럴인, 일종의 편법이에요. 😁